src/Core/src/Hosting/AppHostBuilderExtensions.cs

```diff
@@ -59,5 +61,31 @@ public static IAppHostBuilder UseFonts(this IAppHostBuilder builder)
 			});
 			return builder;
 		}
+
+		public static IAppHostBuilder UseMauiApp<TApp>(this IAppHostBuilder builder)
+			where TApp : MauiApp
+		{
+			builder.ConfigureServices((context, collection) =>
+			{
+				collection.AddSingleton<MauiApp, TApp>();
```

Comment:
I do not know if this is the best thing, but once it is added here then all DI just works. Maybe we need a special place to prevent user overrides. Maybe we don't. We could also add this to a separate place and then inject at the very end so the user cannot override it...